### ISSUE TITLE
ReferenceError: parse_params is not defined

### ISSUE THREAD

**Issue by PavlosIsaris** [marklogic/node-client-api#713]
In the newer NodeJS versions, when trying to `require` the library, the following error get thrown:

```
"ReferenceError: parse_params is not defined",
    "    at node_modules/www-authenticate/lib/parsers.js (/var/task/index.js:18318:18)",
    "    at __require (/var/task/index.js:9:50)",
    "    at node_modules/www-authenticate/lib/www-authenticate.js (/var/task/index.js:18463:19)",
    "    at __require (/var/task/index.js:9:50)",
    "    at node_modules/www-authenticate/index.js (/var/task/index.js:18633:28)",
    "    at __require (/var/task/index.js:9:50)",
    "    at node_modules/marklogic/lib/requester.js (/var/task/index.js:24206:33)",
    "    at __require (/var/task/index.js:9:50)",
    "    at node_modules/marklogic/lib/documents.js (/var/task/index.js:30830:21)",
    "    at __require (/var/task/index.js:9:50)"
```

This is because of the [www-authenticated](https://github.com/randymized/www-authenticate) library.
A relevant [pull request](https://github.com/randymized/www-authenticate/pull/2#issue-1569820441) that solves the issue has already been opened, but not merged yet.
Do you propose any other ways to solve it?

**Comment by anu3990** [comment]
@PavlosIsaris Thank you for the fix.! Currently Node-client uses 0.6.2 version of www-authenticate. Is this fix available in the 0.6.3 version? We can upgrade to 0.6.3.

**Comment by PavlosIsaris** [comment]
@anu3990 As I can see from the [www-authenticate repo](https://github.com/randymized/www-authenticate), it was marked as "archived", meaning that the library is slowly getting abandoned. So, I don't think that there will be a 0.6.3 release. But maybe the author added the fix in the 0.6.2 release. 
Anyway, it will be a good idea to abandon this library, and use another, equivalent one.
